entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/MembershipEntityProvider.java

```diff
@@ -840,11 +854,16 @@ public String createEntity(EntityReference ref, Object entity, Map<String, Objec
 
         checkSiteSecurity(sg.site.getId());
 
+        String[] userAuditString;
+        List<String[]> userAuditList = new ArrayList<>();
+        
         // check for a batch add
         String[] userIds = checkForBatch(params, userId);
-        // now add all the memberships
         String memberId = "";
         String currentUserId = developerHelperService.getCurrentUserId();
+        User user = null;
```

Comment:
I don't think you're using this `User` object outside of the `for` loop below, so I would move it inside that loop scope.